src/manifest/store.rs

```diff
@@ -98,15 +98,26 @@ impl FenceableManifest {
         stored_epoch: fn(&Manifest) -> u64,
         set_epoch: fn(&mut DirtyManifest, u64),
     ) -> Result<Self, SlateDBError> {
-        let local_epoch = stored_epoch(&stored_manifest.manifest) + 1;
-        let mut manifest = stored_manifest.prepare_dirty();
-        set_epoch(&mut manifest, local_epoch);
-        stored_manifest.update_manifest(manifest).await?;
-        Ok(Self {
-            stored_manifest,
-            local_epoch,
-            stored_epoch,
-        })
+        loop {
```

Comment:
[minor] curious if we should have some limit to the number of retries - is there a situation where two machines startup and hit this loop at the same time effectively causing them to keep pushing one another out with `ManifestVersionExists`?

Reply:
Agreed. I filed https://github.com/slatedb/slatedb/issues/571.